optuna/study.py

```diff
@@ -1411,3 +1411,55 @@ def objective(trial):
 
     storage = storages.get_storage(storage)
     return storage.get_all_study_summaries()
+
+
+@experimental("2.8.0")
+class MaxTrialsCallback:
```

Comment:
Note from "offline" discussion: Implement this class in a separate file (e.g. `optuna/_callbacks.py` since `optuna/study.py` is getting big.